lib/root_loader.rb

```diff
@@ -12,30 +14,66 @@ class << self
     # if they don't already exist. 
     def load account, filename
       start_t = Time.now
+      
       Sequel::Model.db.transaction do
         admin_id = "#{account}:user:admin"
         admin = ::Role[admin_id] || ::Role.create(role_id: admin_id)
+        
         if admin_password = ENV['CONJUR_ADMIN_PASSWORD']
           $stderr.puts "Setting 'admin' password"
-          admin_credentials = Credentials[role: admin] || Credentials.create(role: admin)
-          admin_credentials.password = admin_password
-          admin_credentials.save
+          set_admin_password(admin, admin_password)
         end
 
         root_policy_resource = Loader::Types.find_or_create_root_policy(account)
+        policy_ver = policy_version(
+          role: admin, 
+          policy: root_policy_resource, 
+          filename: filename
+        )
 
-        policy_version = PolicyVersion.new role: admin, policy: root_policy_resource, policy_text: File.read(filename)
-        policy_version.policy_filename = filename
-        policy_version.perform_automatic_deletion = true
-        policy_version.delete_permitted = true
-        policy_version.update_permitted = true
-        policy_version.save
+        acc_roles = accepted_roles(policy_ver)
+        created_roles = create_roles(acc_roles)
+
+        $stderr.puts(
+          JSON.pretty_generate(
+            created_roles: created_roles,
+            version: policy_ver.version
+          )
+        )
+
+        $stderr.puts "Loaded policy in #{Time.now - start_t} seconds"
+      end
+    end
+
+    def set_admin_password(admin_role, password)
+      admin_credentials = Credentials[role: admin_role] || Credentials.create(role: admin_role)
+      admin_credentials.password = password
+      admin_credentials.save
+    end
+
+    def create_roles accepted_roles
+      accepted_roles.each_with_object({}) do |role, memo|
+        credentials = Credentials[role: role] || Credentials.create(role: role)
+        role_id = role.id
+        memo[role_id] = { id: role_id, api_key: credentials.api_key }
+      end
+    end
 
-        loader = Loader::Orchestrate.new policy_version
-        loader.load
+    def accepted_roles policy_version
+      loader = Loader::Orchestrate.new policy_version
+      loader.load
+      loader.new_roles.select do |role|
+        %w(user host).member?(role.kind)
       end
-      end_t = Time.now
-      $stderr.puts "Loaded policy in #{end_t - start_t} seconds"
+    end
+
+    def policy_version role:, policy:, filename:
+      policy_version = PolicyVersion.new role: role, policy: policy, policy_text: File.read(filename)
+      policy_version.policy_filename = filename
+      policy_version.perform_automatic_deletion =
+        policy_version.delete_permitted =
+        policy_version.update_permitted = true
+      policy_version.save
```

Comment:
What's happening here?  It looks like something went wrong...

Reply:
Just being naughty and seeing if reek would complain. 🤭 

Was trying a multiple assignment to get around the statement limit.

Reply:
And I would have gotten away with it too if it wasn't for that darn indentation warning! 🐕

Reply:
hahahahaha

This is a clear case where reek may be ignored.  BUT.... the deeper reason it's throwing the warning is because the API for `PolicyVersion.new`  uses the "create and configure" pattern, which is poor and should be avoided.  Instead the constructor should take a hash and return an immutable object.

Also, what does the dog emoji mean?

Reply:
```
[true] * 3
```
😆 😆 

I think it's time to admit defeat and just ignore the CC error.

Reply:
Hah yeah I jokingly suggested `[true] * 3` to Hadar too. 😂 

The dog emoji was a reference to Scooby Doo's "and I would have gotten away with it too if it wasn't for you darn kids!" 😄